c/include/libsbp/cpp/sbp_stdio.h

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) 2020 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SBP_CPP_STDIO_H
+#define SBP_CPP_STDIO_H
+
+#include <fstream>
+
+#include <libsbp/cpp/message_handler.h>
+
+namespace sbp {
+
+class SbpFileReader : public sbp::IReader {
+  public:
+    SbpFileReader(const std::string &file_path)
```

Comment:
Is the general policy to use `const char *` where possible (unless you need to string manipulation functions that `std::string` offers?) 

I was reading `Cases where you might prefer char* over std:string` here: https://www.geeksforgeeks.org/char-vs-stdstring-vs-char-c/

Reply:
For general C++ programming I'd say `std::string` is almost always preferred. But since we're attempting to keep libsbp code more or less portable `const char *` is the lowest common denominator. Like I said before, it's difficult to imagine a situation where libsbp would be used where we're using `std::fstream` without concern but where we wouldn't want to use `std::string`.